engine/src/main/java/org/hibernate/search/engine/impl/ImmutableAnalyzerRegistry.java

```diff
@@ -67,13 +77,19 @@ public AnalyzerReference getAnalyzerReference(Class<?> analyzerClazz) {
 
 	@Override
 	public void close() {
-		close( referencesByClass.values() );
-	}
-
-	private void close(Collection<AnalyzerReference> references) {
-		for ( AnalyzerReference reference : references ) {
+		for ( AnalyzerReference reference : getAllReferences() ) {
 			reference.close();
 		}
 	}
 
+	private List<AnalyzerReference> getAllReferences() {
```

Comment:
not important as it's not an hot code area, but it looks a bit wasteful to add them all to a new collection ?
You could just close them all.

BTW should we invoke each close() method in a try/catch "log & ignore" block?